### ISSUE TITLE
Reduce log verbosity by hiding NO-SOURCE messages

### ISSUE THREAD

**Issue by vlsi** [gradle/gradle#11364]
### Expected Behavior

Gradle should refrain from printing NO-SOURCE messages.
It might make sense to extend the final `X actionable task: Y executed` with `Z tasks having no-source`.

### Current Behavior

NO-SOURCE is a quite common task outcome, and it produces lots of non-actionable log messages.

For instance (see https://github.com/gradle/gradle/issues/10489 )

```
$ ./gradlew baseServices:compileJava --console=verbose
> Task :buildSrc:performance:processResources NO-SOURCE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:build:compileJava NO-SOURCE
> Task :buildSrc:binaryCompatibility:pluginDescriptors UP-TO-DATE
> Task :buildSrc:kotlinDsl:extractPrecompiledScriptPluginPlugins UP-TO-DATE
> Task :buildSrc:plugins:extractPrecompiledScriptPluginPlugins UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
```

^^ The half of the messages is NO-SOURCE, and it makes it compilcated to analyze the log.

Another example:
https://travis-ci.org/apache/jmeter/jobs/611076915

124 `NO-SOURCE`
797 `> Task:`

In other words, 15%=124/797 of all the `> Task` output in Apache JMeter build is related to `NO-SOURCE`.

https://travis-ci.org/gradle/gradle/builds/611516395

1207 `NO-SOURCE`
3530 `> Task:`

So 34% of task lines in Gradle-Travis build are consumed by `NO-SOURCE`

### Context

I would like to reduce the verbosity of the output to make it more actionable.

There's a risk that a user would expect a task to have input data, so there might be needed an option to still produce NO-SOURCE output. So an extra option is probably needed to still show no-source task lines.
